### PR TITLE
ci: Migrate to Melos 3 and Flutter 3.7.1

### PR DESCRIPTION
**Pull request by spydon**
# Description

This PR migrates us to the prerelease version of Melos (3.0.0-dev.0) and our pipelines to Flutter 3.7.1

It's a golden opportunity for us to test the upcoming version of melos and give them any feedback that we might have. :)

The migration instructions can be read here:
https://melos.invertase.dev/guides/migrations

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by erickzanardo — APPROVED** [review]
PR title and description is probaly wrong, I bet you mean flutter 3.7.1 :P

but LGTM!

File: melos.yaml [hunk 1]
```diff
@@ -3,22 +3,17 @@ repository: https://github.com/flame-engine/flame
...
-    usePubspecOverrides: true
-
   version:
-    # Generate commit links in package changelogs.
-    linkToCommits: true
     # Only allow versioning to happen on main branch.
     branch: main
-    # Additionally build a changelog at the root of the workspace.
-    workspaceChangelog: true
+    # Generates a link to a prefilled GitHug release creation page.
```
Comment (st-pasha) [inline melos.yaml:15]: I could use a GitHug right now :)

Comment (spydon) [reply melos.yaml:15]: Haha when your code is so good it feels like a hug 😄

Comment (spydon) [inline melos.yaml:15]:
```suggestion
    # Generates a link to a prefilled GitHub release creation page.
```

**Review by st-pasha — COMMENTED** [review]
Files flat_hex_even.png and flat_hex_odd.png seem to be rendering incorrectly now: some tiles have discolored lines in the middle.

**Comment by spydon** [comment]
> Files flat_hex_even.png and flat_hex_odd.png seem to be rendering incorrectly now: some tiles have discolored lines in the middle.

Hmm, that is weird... Must be some regression with the newest Flutter version then.

**Comment by st-pasha** [comment]
> Must be some regression with the newest Flutter version then.

Should we wait with merging then? I imagine some users wouldn't want to switch to Flutter version that has this kind of bug...

**Comment by spydon** [comment]
> > Must be some regression with the newest Flutter version then.
> 
> Should we wait with merging then? I imagine some users wouldn't want to switch to Flutter version that has this kind of bug...

Us merging this doesn't dictate what version that the user is running, so they would still have that result when running 3.7.0< which most probably do.
I suggest that we merge this and open an issue to investigate that regression more.

**Review by st-pasha — APPROVED** [review]

**Pull request merged**